### PR TITLE
 #292 Exchanging ID Site token for a Stormpath Access Token

### PR DESCRIPTION
**Pull request by dogeared**
Please refer to #376 for previous comments

### CODE REVIEW

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticationRequest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.stormpath.sdk.lang.Assert;
+import com.stormpath.sdk.oauth.IdSiteAuthenticationRequest;
+
+/**
+ * @since 1.0.RC7
+ */
+public class DefaultIdSiteAuthenticationRequest implements IdSiteAuthenticationRequest {
+
+    private final static String grant_type = "id_site_token";
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticationRequest.java]: this should now be `stormpath_token` @lhazlewood - where is this documented?

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/application/ApplicationIT.groovy [hunk 2]
```diff
@@ -30,6 +30,7 @@ import com.stormpath.sdk.application.ApplicationAccountStoreMappingList
 import com.stormpath.sdk.application.Applications
 import com.stormpath.sdk.oauth.AccessToken
 import com.stormpath.sdk.oauth.Authenticators
+import com.stormpath.sdk.oauth.IdSiteAuthenticationRequest
```
Comment (dogeared) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/application/ApplicationIT.groovy]: @maranai - this import is unused.

File: impl/src/main/java/com/stormpath/sdk/impl/application/DefaultApplication.java [hunk 3]
```diff
@@ -857,4 +871,8 @@ public JwtAuthenticator createJwtAuthenticator() {
         return new DefaultJwtAuthenticator(this, getDataStore());
     }
 
+    /* @since 1.0.RC7 */
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/application/DefaultApplication.java]: Should this `1.0.RC8.1`?

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import com.stormpath.sdk.oauth.OauthGrantAuthenticationResult;
+import com.stormpath.sdk.oauth.IdSiteAuthenticationRequest;
+import com.stormpath.sdk.oauth.GrantAuthenticationToken;
+
+/**
+ * @since 1.0.RC8.1
+ */
+public class DefaultIdSiteAuthenticator extends AbstractOauth2Authenticator implements IdSiteAuthenticator {
+
+    final static String OAUTH_TOKEN_PATH = "/oauth/token";
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticator.java]: @maranai - does the work that @mrioan doing on token management impact this?

**Comment by dogeared** [comment]
@maranai - some minor changes and just checking that what @mrioan is working on with token management won't impact anything here.

Assuming that's the case, this PR is approved.

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticator.java (hunk 4 above)
Comment (maranai) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultIdSiteAuthenticator.java]: @dogeared I realized I didn't reply to your comment yesterday.
The answer is no, the endpoint specified in this line refers to the AM endpoint against which the authentication is performed; @mrioan is working on the JSDK endpoint that servlet plugin uses for Oauth authentication.

**Pull request merged**